Added combat potions to TD

changednamedes opened the pull request:
Added the ability to set and use combat/ranging potions (make sure this is in your inventory setup)

changednamedes committed:
Added the ability to set and use combat/ranging potions (make sure this is in your inventory setup)

chsami commented on runelite-client/src/main/java/net/runelite/client/plugins/microbot/zerozero/tormenteddemons/TormentedDemonScript.java:
```diff
@@ -473,6 +476,89 @@ private void teleportToFeroxEnclave() {
+        List<Integer> rangingPotionIds = null;
+        switch (config.rangingPotionType()) {
+            case RANGING:
+                rangingPotionIds = List.of(
```
this is missing normal bastion potion. we have a utility function called: 

`Rs2Potion.getRangePotionsVariants()`

You can use it as:

boolean hasRangePotion = `Rs2Inventory.hasItem(Rs2Potion.getRangePotionsVariants())`

or interact with:

`interact(Rs2Potion.getRangePotionsVariants(), ""Drink")`

chsami commented on runelite-client/src/main/java/net/runelite/client/plugins/microbot/zerozero/tormenteddemons/TormentedDemonScript.java:
```diff
@@ -473,6 +476,89 @@ private void teleportToFeroxEnclave() {
+            }
+        }
+
+        if (rangingPotionIds != null && getPercentageDifference(currentRange, maxRange) < threshold) {
```
Rs2Player also has a bunch of premade methods to check if  divine bastion is still active

```
    public static boolean hasRangingPotionActive() {
        return Microbot.getClient().getBoostedSkillLevel(Skill.RANGED) - 5 > Microbot.getClient().getRealSkillLevel(Skill.RANGED);
    }

    public static boolean hasDivineBastionActive() {
        return divineBastionTime > 0;
    }
```

You should check if you can use any of these methods or maybe slightly adjust them to your needs so we don't have the logic for handling potions spread out in each script.

changednamedes replied:
Should be fine now

changednamedes committed:
Added the below to the Rs2Player Class

hasDivineCombatActive
hasAttackActive
hasStrengthActive
hasDefenseActive

Updated the TD plugin to use these methods to check for potion usage.

changednamedes committed:
Updated the only usage of hasRangingPotionActive (Vorkath) add the old default value, before changing it to a threshold.
